src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentEntity.java

```diff
@@ -294,4 +273,55 @@ public boolean hasSeqNoPrimaryTermProperty() {
 	public ElasticsearchPersistentProperty getSeqNoPrimaryTermProperty() {
 		return seqNoPrimaryTermProperty;
 	}
+
+	private static class IndexCoordinatesResolver {
+
+		private final SpelExpressionParser parser;
```

Comment:
Let's keep the parser as constant so we don't need to create a new instance for each entity. In general, the design could be trimmed down to keeping the parser and a potential `Expression` object (see [`BasicMongoPersistentEntity`](https://github.com/spring-projects/spring-data-mongodb/blob/master/spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/mapping/BasicMongoPersistentEntity.java#L59-L65) for further reference).

Reply:
I moved the IndexCoordinateResolver stuff into SimpleElasticsearchPersistentEntity.  But we need more than one expression when we later introduce the writeIndexname in addition to the current one. That's also the reason I kept the resolve in `getIndexCoordinates()` and did not move it into `getIndexName`-- we'd need to either change that later or have two calls to `resolve`